lib/plugins/aws/utils/request.js

```diff
@@ -0,0 +1,208 @@
+'use strict';
+
+const _ = require('lodash');
+const BbPromise = require('bluebird');
+const memoize = require('memoizee');
+const PromiseQueue = require('promise-queue');
+const sdk = require('aws-sdk');
+const chalk = require('chalk');
+const ServerlessError = require('../../../../lib/serverless-error');
+const log = require('@serverless/utils/log');
+const HttpsProxyAgent = require('https-proxy-agent');
+const https = require('https');
+const fs = require('fs');
+
+// Activate AWS SDK logging
+if (process.env.SLS_DEBUG) {
+  sdk.config.logger = log;
+}
+
+// Use HTTPS Proxy (Optional)
+const proxy =
+  process.env.proxy ||
+  process.env.HTTP_PROXY ||
+  process.env.http_proxy ||
+  process.env.HTTPS_PROXY ||
+  process.env.https_proxy;
+
+const proxyOptions = {};
+if (proxy) {
+  Object.assign(proxyOptions, new URL(proxy));
+}
+
+const ca = process.env.ca || process.env.HTTPS_CA || process.env.https_ca;
+
+let caCerts = [];
+
+if (ca) {
+  // Can be a single certificate or multiple, comma separated.
+  const caArr = ca.split(',');
+  // Replace the newline -- https://stackoverflow.com/questions/30400341
+  caCerts = caCerts.concat(caArr.map((cert) => cert.replace(/\\n/g, '\n')));
+}
+
+const cafile = process.env.cafile || process.env.HTTPS_CAFILE || process.env.https_cafile;
+
+if (cafile) {
+  // Can be a single certificate file path or multiple paths, comma separated.
+  const caPathArr = cafile.split(',');
+  caCerts = caCerts.concat(caPathArr.map((cafilePath) => fs.readFileSync(cafilePath.trim())));
+}
+
+if (caCerts.length > 0) {
+  Object.assign(proxyOptions, {
+    rejectUnauthorized: true,
+    ca: caCerts,
+  });
+}
+
+// Passes also certifications
+if (proxy) {
+  sdk.config.httpOptions.agent = new HttpsProxyAgent(proxyOptions);
+} else if (proxyOptions.ca) {
+  // Update the agent -- http://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/node-registering-certs.html
+  sdk.config.httpOptions.agent = new https.Agent(proxyOptions);
+}
+
+// Configure the AWS Client timeout (Optional).  The default is 120000 (2 minutes)
+const timeout = process.env.AWS_CLIENT_TIMEOUT || process.env.aws_client_timeout;
+if (timeout) {
+  sdk.config.httpOptions.timeout = parseInt(timeout, 10);
+}
+PromiseQueue.configure(BbPromise.Promise);
+const requestQueue = new PromiseQueue(2, Infinity);
+
+const MAX_RETRIES = (() => {
+  const userValue = Number(process.env.SLS_AWS_REQUEST_MAX_RETRIES);
+  return userValue >= 0 ? userValue : 4;
+})();
+
+function awsRequest(service, method, params = {}, serviceOptions = {}) {
```

Comment:
I know it was not outlined in issue description, but as I look into it now. I think it'll be better if if `service` and `serviceOptions` arguments are cumulated into one.

Same as we do it here: https://github.com/serverless/test/blob/bd685dc41a35c39da9d82ccd6fc5d60ec5d4be62/aws-request.js#L21 (note also `getServiceInstance` method).

Reply:
ok, I will change it to match this implementation. It looks better indeed.